frontend/rust-lib/flowy-database2/tests/database/cell_test/test.rs

```diff
@@ -26,40 +24,42 @@ async fn grid_cell_update() {
         continue;
       }
       let cell_changeset = match field_type {
-        FieldType::RichText => "".to_string(),
-        FieldType::Number => "123".to_string(),
-        FieldType::DateTime => make_date_cell_string(123),
+        FieldType::RichText => BoxAny::new("".to_string()),
+        FieldType::Number => BoxAny::new("123".to_string()),
+        FieldType::DateTime => BoxAny::new(DateCellChangeset {
+          date: Some(123),
+          ..Default::default()
+        }),
         FieldType::SingleSelect => {
           let type_option = field
             .get_type_option::<SingleSelectTypeOption>(field.field_type)
             .unwrap();
-          SelectOptionCellChangeset::from_insert_option_id(&type_option.options.first().unwrap().id)
-            .to_cell_changeset_str()
+          BoxAny::new(SelectOptionCellChangeset::from_insert_option_id(
+            &type_option.options.first().unwrap().id,
+          ))
         },
         FieldType::MultiSelect => {
           let type_option = field
             .get_type_option::<MultiSelectTypeOption>(field.field_type)
             .unwrap();
-          SelectOptionCellChangeset::from_insert_option_id(&type_option.options.first().unwrap().id)
-            .to_cell_changeset_str()
+          BoxAny::new(SelectOptionCellChangeset::from_insert_option_id(
+            &type_option.options.first().unwrap().id,
+          ))
         },
-        FieldType::Checklist => ChecklistCellChangeset {
-          insert_options: vec!["new option".to_string()],
+        FieldType::Checklist => BoxAny::new(ChecklistCellChangeset {
+          insert_options: vec![("new option".to_string(), false)],
           ..Default::default()
-        }
-        .to_cell_changeset_str(),
-        FieldType::Checkbox => "1".to_string(),
-        FieldType::URL => "1".to_string(),
-        _ => "".to_string(),
+        }),
+        FieldType::Checkbox => BoxAny::new("1".to_string()),
+        FieldType::URL => BoxAny::new("1".to_string()),
+        _ => BoxAny::new("".to_string()),
       };
 
       scripts.push(UpdateCell {
-        changeset: CellChangesetPB {
-          view_id: test.view_id.clone(),
-          row_id: row_detail.row.id.clone().into(),
-          field_id: field.id.clone(),
-          cell_changeset,
-        },
+        view_id: test.view_id.clone(),
+        field_id: field.id.clone(),
+        row_id: row_detail.row.id.clone(),
+        changeset: cell_changeset,
```

Comment:
You can remove the `BoxAny::new` from all the match arms.
```suggestion
        changeset: Box:Any::new(cell_changeset),
```

Reply:
The `cell_changeset` is of different types depending on the field type. You can take a look at https://github.com/AppFlowy-IO/AppFlowy/pull/4655